rpc/errors.go

```diff
@@ -61,7 +60,11 @@ func tryUnwrapToRPCErr(err error, rpcErrors ...*RPCError) *RPCError {
 			return &nodeErr
 		}
 	}
-	return Err(InternalError, fmt.Sprintln(nodeErr.Code, nodeErr.Message, nodeErr.Data))
+
+	if nodeErr.Code == 0 {
+		return Err(InternalError, err.Error())
+	}
+	return Err(nodeErr.Code, nodeErr.Data)
```

Comment:
Why the change here?

Reply:
Because the previous version was like "If this error isn't one of these passed by the parameter, I'll set it to internal error without checking if it is a valid error of another type".

The code unmarshals the error to an RPCError in the `nodeErr` var so if the "nodeErr.Code" is zero (default value, meaning that it hasn't a code) that means that it is not a valid RPCError, we consider as "Internal error". But, if it is not zero, that means that this error is an existing error, so let's at least pass it to the "Err" function (in this function, the default value is InternalError anyway).

I did this because I faced a situation where the error was "InvalidParams", a valid case in the switch of the "Err" function but the code was hardcoding "InternalError" without even validating it.